lib/api/src/modules/url.ts

```diff
@@ -20,6 +28,26 @@ export interface SubState {
   customQueryParams: QueryParams;
 }
 
+const argsParseOptions = {
+  allowDots: true,
+  delimiter: ';',
+};
+
+const parseArgs = (argsString: string) =>
+  parse(
+    argsString
+      .split(';')
+      .map((part) => part.replace(':', '='))
+      .join(';'),
+    argsParseOptions
+  );
+
+const stringifyArgs = (args: Story['args']) =>
+  stringify(args, { ...argsParseOptions, encode: false })
+    .split(';')
+    .map((part) => part.replace('=', ':'))
+    .join(';');
```

Comment:
I would expect these to be in a separate file and there to be a variety of unit tests ;)